sim/core/spell_mod.go

```diff
@@ -25,6 +26,7 @@ type SpellModConfig struct {
 	KeyValue     string
 	ApplyCustom  SpellModApply
 	RemoveCustom SpellModRemove
+	OnReset      SpellModOnReset
```

Comment:
`ResetCustom` would be more consistent naming within `SpellModConfig`, since this field is ignored except when `config.Kind == SpellMod_Custom`.